### PR TITLE
Customized widgets

### PR DESCRIPTION
**Pull request by furkankurt**
Hello, 

I customized the `MaterialAnimatedSnackBar` and `RectangleAnimatedSnackBar` widgets and updated dependencies for the latest Flutter version.
Also added Duration and Curve for animation.

I bumped the version and added `README.md` for usage.

### CODE REVIEW

File: lib/src/widgets/rectangle_animated_snack_bar.dart [hunk 1]
```diff
@@ -3,73 +3,101 @@ import 'package:flutter/material.dart';
...
     Key? key,
     required this.titleText,
     required this.messageText,
     required this.type,
     required this.brightness,
+    this.infoBackgroundColor = _infoBackgroundColor,
+    this.errorBackgroundColor = _errorBackgroundColor,
+    this.successBackgroundColor = _successBackgroundColor,
+    this.warningBackgroundColor = _warningBackgroundColor,
+    this.infoIcon = _infoIcon,
```
Comment (JulyWitch) [inline lib/src/widgets/rectangle_animated_snack_bar.dart]: For this part, Why don't just get one icon and one color? Or use factories for each type

**Review by JulyWitch — CHANGES_REQUESTED** [review]
Hey Furkan,

I really appreciate your help, Could you please fix the change request?

File: lib/src/widgets/rectangle_animated_snack_bar.dart (hunk 1 above)
Comment (furkankurt) [reply lib/src/widgets/rectangle_animated_snack_bar.dart]: I wasn't able to continue with it.
Apologies for the delay.

**Comment by furkankurt** [comment]
Hello @JulyWitch , can you please review my PR?

**Comment by JulyWitch** [comment]
> Hello @JulyWitch , can you please review my PR?

Yeah sure, Sorry for the delay

**Review by JulyWitch — APPROVED** [review]

**Pull request merged**